### PR TITLE
SPM-7785 create template

### PR DESCRIPTION
**Pull request by amartyushov**

### CODE REVIEW

File: src/main/java/com/signavio/application/mapper/package-info.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Here we should put mappers for entities and dtos, events and messages.
+ * <p>
+ * Only interfaces are allowed here.
+ */
+package com.signavio.application.mapper;
```
Comment (amartyushov) [inline src/main/java/com/signavio/application/mapper/package-info.java:6]: I am not sure about `mapper` package with mapper interfaces to be placed in `application`. Open for discussion.

Comment (amartyushov) [reply src/main/java/com/signavio/application/mapper/package-info.java]: Solved. `mapper` interfaces are under `adapter/common/mapper`

File: doc/architecture.puml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+@startuml
+
+package "Core" {
+	[Domain] as domain
+	[Usecase] as usecase
+}
+
+package "Application" {
+	[AppStarter]
+	[mapper] as mapper
```
Comment (adrianoatsignavio) [inline doc/architecture.puml:10]: you forgot to move the mapper package to adapter

File: src/main/java/com/signavio/template/adapter/messaging/publisher/package-info.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
+ * Here we put implementations of ports with publishing concerns
```
Comment (adrianoatsignavio) [inline src/main/java/com/signavio/template/adapter/messaging/publisher/package-info.java:2]: we could say: "Here we put implementations of ports with **message** publishing concerns"

File: src/main/java/com/signavio/template/application/package-info.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Here we should put classes which are web framework specific.
```
Comment (adrianoatsignavio) [inline src/main/java/com/signavio/template/application/package-info.java:2]: might be that we could create a future service that doesn't hava any web interface at all. i.e. 100% message based system

Comment (adrianoatsignavio) [reply src/main/java/com/signavio/template/application/package-info.java:2]: We could say "Here we put necessary classes to create a stand-alone application"

File: doc/architecture.puml (hunk 2 above)
Comment (amartyushov) [reply doc/architecture.puml:10]: fixed

File: src/main/java/com/signavio/template/adapter/messaging/publisher/package-info.java (hunk 3 above)
Comment (amartyushov) [reply src/main/java/com/signavio/template/adapter/messaging/publisher/package-info.java:2]: fixed

File: src/main/java/com/signavio/template/application/package-info.java (hunk 4 above)
Comment (amartyushov) [reply src/main/java/com/signavio/template/application/package-info.java:2]: fixed

**Pull request merged**